src/main/kotlin/io/kuzzle/sdk/protocol/WebSocket.kt

```diff
@@ -21,6 +21,7 @@ import java.net.ConnectException
 import java.net.SocketException
 import java.util.concurrent.CompletableFuture
 import java.util.concurrent.ConcurrentLinkedQueue
+import java.util.concurrent.atomic.AtomicBoolean
```

Comment:
Why is a simple `boolean` not sufficient?

Reply:
Because we are doing this in a thread `CompletableFuture` which means we need an AtomicBoolean to avoid race conditions with the disconnect method